guardian/prosemirror-elements
# feat: Add helper to programmatically update React components outside of a React context

## What does this change?

Adds a helper function that allows us to programmatically update React components outside of their React context. This is useful when our CMS state informs our elements about what they should be renderering.

See the component itself and its tests for details on usage – hopefully, it's self-documenting, and any feedback to improve either API or docs would be great.

This is one approach to solving this problem, but others exist – definitely approach with an open mind! There are a few advantages to doing things in this way: 
- it doesn't extend the library API in solving what is essentially a UI problem
- it provides an imperative way of updating state (useful for consumers) that only rerenders the parts of the application that depend on that state, so it should be reasonably efficient

## How to test

- Does the new component make sense, conceptually and in use? Is there anything we could do to improve usability?
- Do the tests pass, and cover the functionality of the component?
- Does the new button, `Randomise image role options`, affect the image role options in the Image element as expected?
